Common/tests/Helpers/TestPollsterProvider.cs

```diff
@@ -0,0 +1,164 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2022. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+using ArmoniK.Api.Common.Options;
+using ArmoniK.Core.Adapters.Memory;
+using ArmoniK.Core.Adapters.MongoDB;
+using ArmoniK.Core.Common.gRPC.Services;
+using ArmoniK.Core.Common.Injection;
+using ArmoniK.Core.Common.Pollster;
+using ArmoniK.Core.Common.Pollster.TaskProcessingChecker;
+using ArmoniK.Core.Common.Storage;
+using ArmoniK.Core.Common.Stream.Worker;
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using Mongo2Go;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace ArmoniK.Core.Common.Tests.Helpers;
+
+public class TestPollsterProvider : IDisposable
+{
+  private const           string                   DatabaseName   = "ArmoniK_TestDB";
+  private static readonly ActivitySource           ActivitySource = new("ArmoniK.Core.Common.Tests.TestPollsterProvider");
+  private readonly        WebApplication           app_;
+  private readonly        IMongoClient             client_;
+  private readonly        ILoggerFactory           loggerFactory_;
```

Comment:
Lots of github actions warnings bellow

Reply:
Some are bugged, some are fixed in another pr :
![image](https://user-images.githubusercontent.com/88328270/188316446-31245f37-d645-4c23-81d9-2da2cd18b316.png)